logdna/resource_archive.go

```diff
@@ -0,0 +1,278 @@
+	Apikey             string `json:"apikey,omitempty"`
+	Resourceinstanceid string `json:"resourceinstanceid,omitempty"`
+	Accountname        string `json:"accountname,omitempty"`
+	Accountkey         string `json:"accountkey,omitempty"`
+	Projectid          string `json:"projectid,omitempty"`
+	Space              string `json:"space,omitempty"`
+	Accesskey          string `json:"accesskey,omitempty"`
+	Secretkey          string `json:"secretkey,omitempty"`
+	Authurl            string `json:"authurl,omitempty"`
+	Expires            int    `json:"expires,omitempty"`
+	Username           string `json:"username,omitempty"`
+	Password           string `json:"password,omitempty"`
+	Tenantname         string `json:"tenantname,omitempty"`
+}
+
+func resourceArchiveConfigCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	pc := m.(*providerConfig)
+	c := archiveConfig{
+		Integration:        d.Get("integration").(string),
+		Bucket:             d.Get("bucket").(string),
+		Endpoint:           d.Get("endpoint").(string),
+		Apikey:             d.Get("apikey").(string),
+		Resourceinstanceid: d.Get("resourceinstanceid").(string),
+		Accountname:        d.Get("accountname").(string),
+		Accountkey:         d.Get("accountkey").(string),
+		Projectid:          d.Get("projectid").(string),
+		Space:              d.Get("space").(string),
+		Accesskey:          d.Get("accesskey").(string),
+		Secretkey:          d.Get("secretkey").(string),
+		Authurl:            d.Get("authurl").(string),
+		Expires:            d.Get("expires").(int),
+		Username:           d.Get("username").(string),
+		Password:           d.Get("password").(string),
+		Tenantname:         d.Get("tenantname").(string),
+	}
+
+	req := newRequestConfig(
+		pc,
+		"POST",
+		"/v1/config/archiving",
+		c,
+	)
+
+	body, err := req.MakeRequest()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cn := archiveConfig{}
+	err = json.Unmarshal(body, &cn)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(archiveConfigID)
+	appendError(d.Set("integration", cn.Integration), &diags)
+	appendError(d.Set("bucket", cn.Bucket), &diags)
+	appendError(d.Set("endpoint", cn.Endpoint), &diags)
+	appendError(d.Set("apikey", cn.Apikey), &diags)
+	appendError(d.Set("resourceinstanceid", cn.Resourceinstanceid), &diags)
+	appendError(d.Set("accountname", cn.Accountname), &diags)
+	appendError(d.Set("accountkey", cn.Accountkey), &diags)
+	appendError(d.Set("projectid", cn.Projectid), &diags)
+	appendError(d.Set("space", cn.Space), &diags)
+	appendError(d.Set("accesskey", cn.Accesskey), &diags)
+	appendError(d.Set("secretkey", cn.Secretkey), &diags)
+	appendError(d.Set("authurl", cn.Authurl), &diags)
+	appendError(d.Set("expires", cn.Expires), &diags)
+	appendError(d.Set("username", cn.Username), &diags)
+	appendError(d.Set("password", cn.Password), &diags)
+	appendError(d.Set("tenantname", cn.Tenantname), &diags)
+
+	return diags
```

Comment:
Nit: I'm not how important it is to follow Terraform's "best practice" here since this aligns with the streaming config resource but we _could_ shorten lines[73-91] here to
```golang
d.SetId(archiveConfigID)

return resourceArchiveConfigRead(ctx, d, m)
```

Reply:
I believe since the POST response has the created object, we do not need an additional GET call.

Reply:
Fine with me